<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TabTools</class>
 <widget class="QWidget" name="TabTools">
  <property name="geometry">
   <rect><x>0</x><y>0</y><width>900</width><height>600</height></rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_main">
   <property name="leftMargin"><number>4</number></property>
   <property name="topMargin"><number>4</number></property>
   <property name="rightMargin"><number>4</number></property>
   <property name="bottomMargin"><number>4</number></property>
   
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable"><bool>true</bool></property>
     <property name="frameShape"><enum>QFrame::NoFrame</enum></property>
     <widget class="QWidget" name="scrollContent">
      <layout class="QHBoxLayout" name="mainHLayout">
       <property name="spacing"><number>8</number></property>
       
       <!-- ‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê -->
       <!-- COLUMN 1: Convert + Compare                                 -->
       <!-- ‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê -->
       <item>
        <layout class="QVBoxLayout" name="col1">
         <property name="spacing"><number>6</number></property>
         
         <!-- CONVERT -->
         <item>
          <widget class="QGroupBox" name="groupConvert">
           <property name="title"><string>Convert</string></property>
           <property name="styleSheet"><string>QGroupBox { font-weight: bold; }</string></property>
           <layout class="QVBoxLayout">
            <property name="spacing"><number>4</number></property>
            <item>
             <layout class="QHBoxLayout">
              <item><widget class="QLabel"><property name="text"><string>Source:</string></property></widget></item>
              <item>
               <widget class="QLineEdit" name="editConvertSource">
                <property name="placeholderText"><string>Input file...</string></property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnBrowseConvertSource">
                <property name="text"><string>...</string></property>
                <property name="maximumWidth"><number>30</number></property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout">
              <item><widget class="QLabel"><property name="text"><string>Target:</string></property></widget></item>
              <item>
               <widget class="QLineEdit" name="editConvertTarget">
                <property name="placeholderText"><string>Output file...</string></property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnBrowseConvertTarget">
                <property name="text"><string>...</string></property>
                <property name="maximumWidth"><number>30</number></property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout">
              <item><widget class="QLabel"><property name="text"><string>From:</string></property></widget></item>
              <item>
               <widget class="QComboBox" name="comboConvertFrom">
                <item><property name="text"><string>Auto-Detect</string></property></item>
                <item><property name="text"><string>ADF</string></property></item>
                <item><property name="text"><string>D64</string></property></item>
                <item><property name="text"><string>G64</string></property></item>
                <item><property name="text"><string>SCP</string></property></item>
                <item><property name="text"><string>HFE</string></property></item>
                <item><property name="text"><string>IMG</string></property></item>
                <item><property name="text"><string>IMD</string></property></item>
               </widget>
              </item>
              <item><widget class="QLabel"><property name="text"><string>To:</string></property></widget></item>
              <item>
               <widget class="QComboBox" name="comboConvertTo">
                <item><property name="text"><string>ADF</string></property></item>
                <item><property name="text"><string>D64</string></property></item>
                <item><property name="text"><string>G64</string></property></item>
                <item><property name="text"><string>SCP</string></property></item>
                <item><property name="text"><string>HFE</string></property></item>
                <item><property name="text"><string>IMG</string></property></item>
                <item><property name="text"><string>IMD</string></property></item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="btnConvert">
              <property name="text"><string>üîÑ Convert</string></property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         
         <!-- COMPARE -->
         <item>
          <widget class="QGroupBox" name="groupCompare">
           <property name="title"><string>Compare</string></property>
           <property name="styleSheet"><string>QGroupBox { font-weight: bold; }</string></property>
           <layout class="QVBoxLayout">
            <property name="spacing"><number>4</number></property>
            <item>
             <layout class="QHBoxLayout">
              <item><widget class="QLabel"><property name="text"><string>File A:</string></property></widget></item>
              <item>
               <widget class="QLineEdit" name="editCompareA">
                <property name="placeholderText"><string>First file...</string></property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnBrowseCompareA">
                <property name="text"><string>...</string></property>
                <property name="maximumWidth"><number>30</number></property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout">
              <item><widget class="QLabel"><property name="text"><string>File B:</string></property></widget></item>
              <item>
               <widget class="QLineEdit" name="editCompareB">
                <property name="placeholderText"><string>Second file...</string></property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnBrowseCompareB">
                <property name="text"><string>...</string></property>
                <property name="maximumWidth"><number>30</number></property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout">
              <item><widget class="QLabel"><property name="text"><string>Mode:</string></property></widget></item>
              <item>
               <widget class="QComboBox" name="comboCompareMode">
                <item><property name="text"><string>Binary (byte-by-byte)</string></property></item>
                <item><property name="text"><string>Sector</string></property></item>
                <item><property name="text"><string>Track</string></property></item>
                <item><property name="text"><string>Flux timing</string></property></item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout">
              <item>
               <widget class="QCheckBox" name="checkIgnoreHeaders">
                <property name="text"><string>Ignore headers</string></property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="checkShowHex">
                <property name="text"><string>Show hex diff</string></property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="btnCompare">
              <property name="text"><string>üîç Compare</string></property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         
         <item><spacer><property name="orientation"><enum>Qt::Vertical</enum></property></spacer></item>
        </layout>
       </item>
       
       <!-- ‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê -->
       <!-- COLUMN 2: Repair + Analyze                                  -->
       <!-- ‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê -->
       <item>
        <layout class="QVBoxLayout" name="col2">
         <property name="spacing"><number>6</number></property>
         
         <!-- REPAIR -->
         <item>
          <widget class="QGroupBox" name="groupRepair">
           <property name="title"><string>Repair</string></property>
           <property name="styleSheet"><string>QGroupBox { font-weight: bold; }</string></property>
           <layout class="QVBoxLayout">
            <property name="spacing"><number>4</number></property>
            <item>
             <layout class="QHBoxLayout">
              <item><widget class="QLabel"><property name="text"><string>File:</string></property></widget></item>
              <item>
               <widget class="QLineEdit" name="editRepairFile">
                <property name="placeholderText"><string>File to repair...</string></property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnBrowseRepair">
                <property name="text"><string>...</string></property>
                <property name="maximumWidth"><number>30</number></property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel"><property name="text"><string>Repairs:</string></property></widget>
            </item>
            <item>
             <layout class="QGridLayout">
              <item row="0" column="0">
               <widget class="QCheckBox" name="checkFixChecksum">
                <property name="text"><string>Fix checksums</string></property>
                <property name="checked"><bool>true</bool></property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QCheckBox" name="checkFixBAM">
                <property name="text"><string>Fix BAM/FAT</string></property>
                <property name="checked"><bool>true</bool></property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QCheckBox" name="checkFixHeaders">
                <property name="text"><string>Fix headers</string></property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QCheckBox" name="checkRecoverDeleted">
                <property name="text"><string>Recover deleted</string></property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QCheckBox" name="checkFillBadSectors">
                <property name="text"><string>Fill bad sectors</string></property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QCheckBox" name="checkBackup">
                <property name="text"><string>Create backup</string></property>
                <property name="checked"><bool>true</bool></property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="btnRepair">
              <property name="text"><string>üîß Repair</string></property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         
         <!-- ANALYZE -->
         <item>
          <widget class="QGroupBox" name="groupAnalyze">
           <property name="title"><string>Analyze</string></property>
           <property name="styleSheet"><string>QGroupBox { font-weight: bold; }</string></property>
           <layout class="QVBoxLayout">
            <property name="spacing"><number>4</number></property>
            <item>
             <layout class="QHBoxLayout">
              <item><widget class="QLabel"><property name="text"><string>File:</string></property></widget></item>
              <item>
               <widget class="QLineEdit" name="editAnalyzeFile">
                <property name="placeholderText"><string>File to analyze...</string></property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnBrowseAnalyze">
                <property name="text"><string>...</string></property>
                <property name="maximumWidth"><number>30</number></property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QGridLayout">
              <item row="0" column="0">
               <widget class="QCheckBox" name="checkDetectFormat">
                <property name="text"><string>Detect format</string></property>
                <property name="checked"><bool>true</bool></property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QCheckBox" name="checkDetectProtection">
                <property name="text"><string>Detect protection</string></property>
                <property name="checked"><bool>true</bool></property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QCheckBox" name="checkShowDirectory">
                <property name="text"><string>Show directory</string></property>
                <property name="checked"><bool>true</bool></property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QCheckBox" name="checkShowBadSectors">
                <property name="text"><string>Show bad sectors</string></property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QCheckBox" name="checkCalcHashes">
                <property name="text"><string>Calculate hashes</string></property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QCheckBox" name="checkHexDump">
                <property name="text"><string>Hex dump</string></property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="btnAnalyze">
              <property name="text"><string>üìä Analyze</string></property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         
         <item><spacer><property name="orientation"><enum>Qt::Vertical</enum></property></spacer></item>
        </layout>
       </item>
       
       <!-- ‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê -->
       <!-- COLUMN 3: Batch + Quick Tools                               -->
       <!-- ‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê‚ïê -->
       <item>
        <layout class="QVBoxLayout" name="col3">
         <property name="spacing"><number>6</number></property>
         
         <!-- BATCH -->
         <item>
          <widget class="QGroupBox" name="groupBatch">
           <property name="title"><string>Batch</string></property>
           <property name="styleSheet"><string>QGroupBox { font-weight: bold; }</string></property>
           <layout class="QVBoxLayout">
            <property name="spacing"><number>4</number></property>
            <item>
             <layout class="QHBoxLayout">
              <item><widget class="QLabel"><property name="text"><string>Folder:</string></property></widget></item>
              <item>
               <widget class="QLineEdit" name="editBatchFolder">
                <property name="placeholderText"><string>Input folder...</string></property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnBrowseBatch">
                <property name="text"><string>...</string></property>
                <property name="maximumWidth"><number>30</number></property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout">
              <item><widget class="QLabel"><property name="text"><string>Filter:</string></property></widget></item>
              <item>
               <widget class="QLineEdit" name="editBatchFilter">
                <property name="text"><string>*.d64;*.g64;*.adf</string></property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout">
              <item><widget class="QLabel"><property name="text"><string>Action:</string></property></widget></item>
              <item>
               <widget class="QComboBox" name="comboBatchAction">
                <item><property name="text"><string>Convert</string></property></item>
                <item><property name="text"><string>Analyze</string></property></item>
                <item><property name="text"><string>Validate</string></property></item>
                <item><property name="text"><string>Repair</string></property></item>
                <item><property name="text"><string>Calculate hashes</string></property></item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout">
              <item>
               <widget class="QCheckBox" name="checkBatchSubfolders">
                <property name="text"><string>Include subfolders</string></property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="checkBatchLog">
                <property name="text"><string>Create log</string></property>
                <property name="checked"><bool>true</bool></property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout">
              <item>
               <widget class="QPushButton" name="btnBatchStart">
                <property name="text"><string>‚ñ∂ Start Batch</string></property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btnBatchStop">
                <property name="text"><string>‚èπ Stop</string></property>
                <property name="enabled"><bool>false</bool></property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         
         <!-- QUICK TOOLS -->
         <item>
          <widget class="QGroupBox" name="groupQuickTools">
           <property name="title"><string>Quick Tools</string></property>
           <property name="styleSheet"><string>QGroupBox { font-weight: bold; }</string></property>
           <layout class="QGridLayout">
            <property name="verticalSpacing"><number>4</number></property>
            <item row="0" column="0">
             <widget class="QPushButton" name="btnHexView">
              <property name="text"><string>üî¢ Hex View</string></property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QPushButton" name="btnTrackView">
              <property name="text"><string>üìÄ Track View</string></property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QPushButton" name="btnFluxView">
              <property name="text"><string>üìà Flux View</string></property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QPushButton" name="btnSectorEdit">
              <property name="text"><string>‚úèÔ∏è Sector Edit</string></property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QPushButton" name="btnCreateBlank">
              <property name="text"><string>üíæ Create Blank</string></property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QPushButton" name="btnDiskInfo">
              <property name="text"><string>‚ÑπÔ∏è Disk Info</string></property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         
         <item><spacer><property name="orientation"><enum>Qt::Vertical</enum></property></spacer></item>
        </layout>
       </item>
       
      </layout>
     </widget>
    </widget>
   </item>
   
   <!-- OUTPUT / RESULTS -->
   <item>
    <widget class="QGroupBox" name="groupOutput">
     <property name="title"><string>Output</string></property>
     <property name="styleSheet"><string>QGroupBox { font-weight: bold; }</string></property>
     <property name="maximumHeight"><number>150</number></property>
     <layout class="QVBoxLayout">
      <item>
       <widget class="QPlainTextEdit" name="textOutput">
        <property name="readOnly"><bool>true</bool></property>
        <property name="plainText"><string>Ready.</string></property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout">
        <item>
         <widget class="QProgressBar" name="progressBar">
          <property name="value"><number>0</number></property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnClearOutput">
          <property name="text"><string>Clear</string></property>
          <property name="maximumWidth"><number>60</number></property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnSaveOutput">
          <property name="text"><string>Save...</string></property>
          <property name="maximumWidth"><number>60</number></property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
